samples/EventApi/Consumer.Tests/EventsApiConsumerTestsV3.cs

```diff
@@ -0,0 +1,224 @@
+                });
+
+            await pact.VerifyAsync(async ctx =>
+            {
+                var client = new EventsApiClient(ctx.MockServerUri, Token);
+
+                var result = await client.GetEventById(expected.EventId);
+
+                result.Should().BeEquivalentTo(expected);
+            });
+        }
+
+        [Fact]
+        public async Task GetEventsByType_WhenOneEventWithTheTypeExists_ReturnsEvent()
+        {
+            //Arrange
+            const string eventType = "DetailsView";
+
+            pact
+                .UponReceiving($"a request to retrieve events with type '{eventType}'")
+                    .Given($"there is one event with type '{eventType}'")
+                    .WithRequest(HttpMethod.Get, "/events")
+                    .WithQuery("type", eventType)
+                    .WithHeader("Accept", "application/json")
+                    .WithHeader("Authorization", $"Bearer {Token}")
+                .WillRespond()
+                    .WithStatus(200)
+                    .WithHeader("Content-Type", "application/json; charset=utf-8")
+                    .WithJsonBody(new[]
+                    {
+                        new
+                        {
+                            eventType
+                        }
+                    });
+
+            await pact.VerifyAsync(async ctx =>
+            {
+                var client = new EventsApiClient(ctx.MockServerUri, Token);
+
+                var result = await client.GetEventsByType(eventType);
+
+                result.Should().OnlyContain(e => e.EventType == eventType);
+            });
+        }
+
+        #region V3 support
+
+        [Fact]
+        public async Task GetAllEvents_Given_Multiple_ProviderStates_WhenCalled_ReturnsAllEvents()
+        {
+            var expected = new[]
+            {
+                new Event
+                {
+                    EventId = Guid.Parse("45D80D13-D5A2-48D7-8353-CBB4C0EAABF5"),
+                    Timestamp = DateTime.Parse("2014-06-30T01:37:41.0660548"),
+                    EventType = "SearchView"
+                },
+                new Event
+                {
+                    EventId = Guid.Parse("83F9262F-28F1-4703-AB1A-8CFD9E8249C9"),
+                    Timestamp = DateTime.Parse("2014-06-30T01:37:52.2618864"),
+                    EventType = "DetailsView"
+                },
+                new Event
+                {
+                    EventId = Guid.Parse("3E83A96B-2A0C-49B1-9959-26DF23F83AEB"),
+                    Timestamp = DateTime.Parse("2014-06-30T01:38:00.8518952"),
+                    EventType = "SearchView"
+                }
+            };
+
+            pact
+                .UponReceiving("a request to retrieve all events with multiple provider states")
+                .Given("there are events with id '45D80D13-D5A2-48D7-8353-CBB4C0EAABF5'")
```

Comment:
You are right, I wanted to illustrate we could write different instructions but the way it'd be used is for instance to mock another API or component that has nothing to do with the events.

How about something as simple as this :
                .UponReceiving("a request to retrieve all events with multiple provider states")
                .Given("there are events with id '45D80D13-D5A2-48D7-8353-CBB4C0EAABF5'")
                .Given("another provider state")
                
Otherwise, we could add a component in the API sample calling an external API and use the provider state to mock it (overkill?)

Reply:
Did it like this to illustrate the point of having multiple provider states:

            pact
                .UponReceiving("a request to retrieve all events with multiple provider states")
                .Given("several events have already been created")      //Can create events on the provider end
                .Given("all events have been processed")                //Can set other event information such as execution date and status on the provider end
                .WithRequest(HttpMethod.Get, "/events")
                .WithHeader("Accept", "application/json")
                .WithHeader("Authorization", $"Bearer {Token}")
                .WillRespond()
                .WithStatus(HttpStatusCode.OK)
                .WithHeader("Content-Type", "application/json; charset=utf-8")
                .WithJsonBody(expected);